### PR TITLE
stats: support auto analyze table when modify/count is too high

### PR DESCRIPTION
**Pull request by fipped**
This PR supports analyze table automatically when `modify/count` is greater than `auto-analyze-ratio` which set in the configuration. If `auto-analyze-ratio` is not set or set to a negative number, auto analyze won't run.
PTAL @coocood @lamxTyler

### CODE REVIEW

**Comment by coocood** [comment]
@fipped 
It's not safe to set this value to 0.
We should verify that the ratio is a positive value.
The default value can be set to a very high one, like `1e100`.

**Comment by zhexuany** [comment]
@fipped please fix ci. Any variable or function which is exported has to come with some comments.

File: config/config.go [hunk 1]
```diff
@@ -258,6 +259,7 @@ var defaultConf = Config{
 		FeedbackProbability: 0,
 		QueryFeedbackLimit:  1024,
 		PseudoEstimateRatio: 0.7,
+		AutoAnalyzeRatio:    1e100,
```
Comment (zz-jason) [inline config/config.go]: This number is still unsafe, I think we should use another special one to explicitly specify that auto analyze is turned off.

Comment (zz-jason) [reply config/config.go]: why not just set it to -1 ?

Comment (coocood) [reply config/config.go]: No, it is safe.
-1 make the logic complex.

File: statistics/update.go [hunk 2]
```diff
@@ -376,7 +377,7 @@ func needAnalyzeTable(tbl *Table, limit time.Duration) bool {
 	if time.Since(t) < limit {
 		return false
 	}
-	if AutoAnalyzeRatio >= 0 && float64(tbl.ModifyCount)/float64(tbl.Count) > AutoAnalyzeRatio {
+	if AutoAnalyzeRatio > 0 && float64(tbl.ModifyCount)/float64(tbl.Count) > AutoAnalyzeRatio {
```
Comment (coocood) [inline statistics/update.go]: We make sure the `AutoAnalyzeRatio` is greater than 0, then we can remove the `> 0` check.

File: config/config.toml.example [hunk 3]
```diff
@@ -143,6 +143,11 @@ query-feedback-limit = 1024
 # row count in statistics of a table is greater than it.
 pseudo-estimate-ratio = 0.7
 
+# Auto analyze will run if the ratio between the modify count and
+# the row count of a table is greater than it. It should be a positive
+# float number, otherwise auto analyze won't run.
```
Comment (zz-jason) [inline config/config.toml.example]: how about adding "set to 0.0 to disable auto analyze" ?

Comment (fipped) [reply config/config.toml.example]: I think the last sentence implies this.

**Comment by coocood** [comment]
LGTM

File: config/config.toml.example (hunk 3 above)
Comment (zz-jason) [reply config/config.toml.example]: It's better to explicitly specify this important feature other than imply it.

**Comment by coocood** [comment]
@zz-jason PTAL

**Pull request merged**